projects/plugins/boost/app/modules/cache/cache.php

```diff
@@ -0,0 +1,168 @@
+	 * The filter boost_cache_cacheable can be used to override this.
+	 *
+	 * @return bool
+	 */
+	public function is_cacheable() {
+		if ( ! apply_filters( 'boost_cache_cacheable', $this->request_uri ) ) {
+			return false;
+		}
+
+		if ( $this->is_backend() ) {
+			return false;
+		}
+
+		if ( ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] === 'POST' ) ) {
+			return false;
+		}
+
+		if ( isset( $_SERVER['HTTP_ACCEPT'] ) && strpos( $_SERVER['HTTP_ACCEPT'], 'text/html' ) === false ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			return false;
+		}
+
+		return true;
+	}
+
+	/*
+	 * Returns a key to identify the visitor's cache file.
+	 * It is based on the REQUEST_URI, the cookies and the page GET parameters.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the cache key. Defaults to the current request uri.
+	 *
+	 * @return string
+	 */
+	private function calculate_cache_key( $request_uri = '' ) {
+		if ( $request_uri === '' ) {
+			$request_uri = $this->request_uri;
+		}
+		$cookies = isset( $_COOKIE ) ? $_COOKIE : array();
+		$get     = isset( $_GET ) ? $_GET : array(); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+
+		$key_components = apply_filters(
+			'boost_cache_key_components',
+			array(
+				'cookies'     => $cookies,
+				'request_uri' => $request_uri,
+				'get'         => $get,
+			)
+		);
+
+		return md5( json_encode( $key_components ) ); // phpcs:ignore WordPress.WP.AlternativeFunctions.json_encode_json_encode
+	}
+
+	/*
+	 * Returns a key to identify the visitor's cache file.
+	 * It uses calculate_cache_key() to calculate the key.
+	 * Without a parameter, it will use the current request uri, and cache that
+	 * value in $this->cache_key.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the cache key. Defaults to the current request uri.
+	 * @return string
+	 */
+	public function cache_key( $request_uri = '' ) {
+		if ( $request_uri !== '' ) {
+			return $this->calculate_cache_key( $request_uri );
+		}
+
+		if ( ! $this->cache_key ) {
+			$this->cache_key = $this->calculate_cache_key();
+		}
+		return $this->cache_key;
+	}
+
+	/*
+	 * Returns a key to identify the path to the visitor's cache file.
+	 * Without a parameter it uses the current request uri, and caches that in
+	 * $this->path_key.
+	 *
+	 * @param string $request_uri (optional) The request uri to use to calculate the path key. Defaults to the current request uri.
+	 * @return string
+	 */
+	public function path_key( $request_uri = '' ) {
+		if ( $request_uri !== '' ) {
+			return md5( $request_uri );
+		}
+
+		if ( ! $this->path_key ) {
+			$this->path_key = md5( $this->request_uri );
+		}
+		return $this->path_key;
+	}
+
+	public function get() {
+		return false;
+	}
+
+	public function set( $data ) {
+		echo esc_html( $data );
+		return false;
+	}
```

Comment:
Won't `esc_html` escape all the HTML tags in the output? Also seems like a good case to use an abstract method.

Reply:
Same as for get(). I'll make it an abstract function.